src/services/utxoForAddress.ts

```diff
@@ -22,6 +22,7 @@ const utxoForAddressQuery = `
   JOIN block
     on block.id = tx.block_id
   WHERE tx_in.tx_in_id IS NULL
+    and tx.valid_contract
```

Comment:
Ah, I think the check here will have to be a bit more complicated than this. You need to:

1. NOT include any outputs from any invalid tx 
2. NOT exlude outputs if they have been attempted to be spent in an invalid tx 
3. DO exlude outputs if they have been used as a collateral input in an invalid tx

So current logic is: all outputs to one of the addresses, minus any of these outputs that has been in a tx input.

New logic should be something like: all outputs to one of the addresses ONLY from valid txs, minus any of these outputs that has been in a VALID tx input, minus any of these outputs that has been in an INVALID tx COLLATERAL input